### PR TITLE
Anonymizing unknown functions

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2210.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs [hunk 1]
```diff
@@ -98,7 +96,7 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
             foreach (var udf in uDFs)
             {
                 var udfName = udf.Ident.Name;
-                if (_restrictedUDFNames.Contains(udfName) || texlFunctionSet.AnyWithName(udfName) || BuiltinFunctionsCore._library.AnyWithName(udfName) || BuiltinFunctionsCore.OtherKnownFunctions.Contains(udfName))
+                if (_restrictedUDFNames.Contains(udfName) || texlFunctionSet.AnyWithName(udfName) || BuiltinFunctionsCore.IsKnownPublicFunction(udfName))
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs:99]: >BuiltinFunctionsCore.IsKnownPublicFunction(udfName) [](http://example.com/codeflow?start=101&length=51)

@VamsiModem Now this is also checking `_featureGateFunctions`. Do you see a potential problem?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by MikeStall** [comment]
>                 }

Actually - this original logic seems  wrong.
It means adding any new builtin function is a breaking change! We should followup separately.




---
Refers to: src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs:103 in aa1dd48. [](commit_id = aa1dd4848e9454aafbe2fa99c8481edc687011f8, deletion_comment = False)

**Review by MikeStall — DISMISSED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**